libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/LocationEngineWrapper.kt

```diff
@@ -0,0 +1,79 @@
+package com.mapbox.navigation.core.trip.session
+
+import android.annotation.SuppressLint
+import android.app.PendingIntent
+import android.os.Looper
+import com.mapbox.android.core.location.LocationEngine
+import com.mapbox.android.core.location.LocationEngineCallback
+import com.mapbox.android.core.location.LocationEngineRequest
+import com.mapbox.android.core.location.LocationEngineResult
+import java.lang.Exception
+
+/**
+ * LocationEngine does not have a method to cancel get last location task,
+ * as it has `removeLocationUpdates` for `requestLocationUpdates`.
+ * This may lead to getLastLocation callback firing after `stopLocationUpdates` was invoked,
+ * which is not expected behaviour in context of Mapbox trip sessions.
+ * This class introduces an ability to cancel get last location task for a particular callback.
+ * The issue is reproducible with `ReplayLocationTest#replay_session_locations_do_not_contain_locations_from_previous_session`.
+ */
+internal class LocationEngineWrapper(
```

Comment:
```suggestion
/**
 * LocationEngine does not have a method to cancel get last location task,
 * as it has `removeLocationUpdates` for `requestLocationUpdates`.
 * This may lead to getLastLocation callback firing after `stopLocationUpdates` was invoked,
 * which is not expected behaviour in context of Mapbox trip sessions.
 * This class introduces an ability to cancel get last location task for a particular callback.
 * The issue is reproducible with `ReplayLocationTest#replay_session_locations_do_not_contain_locations_from_previous_session`.
 */
internal class CancellableLocationEngine(
```
Maybe highlight the purpose in the name?

Reply:
Thought about this name, but IMO it's a bit misleading. We only add the ability to cancel `getLastLocation` task. And the use case where it's required is not very common. Everything else is cancellable already.
But since you've also thought of it (and it's anyway better than wrapper), will rename.